### PR TITLE
Add new Podman package versions and fix patch

### PR DESCRIPTION
**Pull request by carsonwoods**
This is a small PR which adds many new versions of Podman as well as correcting the patch version range to reflect when the patch was merged into Podman as an upstreamed fix.

### CODE REVIEW

**Comment by alecbcs** [comment]
Hey @carsonwoods thanks for the PR! Do you need all of the latest versions of Podman for your use case or would it be enough to add the latest version? (v4.5.1)

**Comment by carsonwoods** [comment]
Not necessarily, though it could be useful to have intermediary versions. At the site I'm working with, the system-Podman isn't installed through Spack and the administrators don't always choose the latest version when upgrading for various reasons. Ideally, I'd like to be able to install any version I need via Spack so that I could match their version on my own system (or install a different minor version if I'm testing something).

That being said, I could always install a different version manually or without a checksummed version if adding more versions isn't preferred.

**Review by tldahlgren — DISMISSED** [review]
Confirmed the seven new version sha256, rest LGTM but deferring merge decision to give @bernhardkaindl time to review

File: var/spack/repos/builtin/packages/podman/package.py
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/podman/package.py]: @carsonwoods I don't use the spack-built podman currently and have many other projects in the works. From your comments, you are actively using it, and it looks like you might be better suited as maintainer for the podman package than me. Alternatively, you don't want to be the maintainer, I'd like to nominate @alecbcs (in case he agrees)

Would you agree to change the maintainer field to you?

```suggestion
    maintainers("alecbcs")

    version("4.5.1", sha256="ee2c8b02b7fe301057f0382637b995a9c6c74e8d530692d6918e4c509ade6e39")
```
Also, having red the comment from alecbcs and my previous experience improving/fixing `spack` builds (e.g. of packages like `python`, `gcc`, `go`, `autoconf`, `automake`, etc:

From that experience gained, I'd recommend that it would be advisable to add version checksums only for versions which are actively used and should be maintained to work (e.g. for any build or security fixes).

Therefore, to not overdo maintenance work (not spend work on versions which aren't actively used), I propose the only add the current version (or more specifically, the versions that you really need).

Comment (alecbcs) [reply var/spack/repos/builtin/packages/podman/package.py]: > I'd like to nominate @alecbcs (in case he agrees)

I'd be happy to take on Podman as a maintainer. @bernhardkaindl feel free to add me to your suggestion.

Comment (carsonwoods) [reply var/spack/repos/builtin/packages/podman/package.py]: I think @alecbcs would be a great choice as well. I don't mind doing it, but I'm about to start a Ph.D. program and subsequently will be transitioning away from my Podman-related projects at my current job in the next month or two. As such, my experience with the package might become outdated as time goes on. 

I'd be happy to take Alec's suggestion to remove all but the latest version until someone has a use case for a specific older version when (or if) that ever happens. 

I'll go ahead and make the changes, and leave it to you both to give the final approval. Thanks for taking the time to review the PR!

**Comment by carsonwoods** [comment]
@alecbcs @bernhardkaindl I've gone ahead and made your suggested changes. Let me know what you both think. Thanks again for the reviews!

**Review by bernhardkaindl — APPROVED** [review]
Perfect, the comments are applied! Thank you @carsonwoods and @alecbcs!

**Comment by alecbcs** [comment]
@carsonwoods Thanks for the refactor and congrats on starting your PhD!

**Pull request merged**
